src/amount_ext.rs

```diff
@@ -21,10 +25,53 @@ impl FromCliStr for Amount {
     }
 }
 
+fn parse_suffix(string: &str) -> Option<Result<Amount, anyhow::Error>> {
+    let final_value: f32;
+    if void_denominations(string) {
+        match string.rfind(char::is_numeric) {
+            Some(i) => {
+                let user_value = &string[..=i].parse::<f32>().unwrap();
+                if check_k(string) {
+                    final_value = 1_000 as f32;
+                } else if check_m(string) {
+                    final_value = 1_000_000 as f32;
+                } else {
+                    return Some(Err(anyhow!(
+                        "'{}' is missing a denomination or a suffix, like k or m",
+                        string
+                    )));
+                }
+                return Some(Ok(Amount::from_sat((user_value * final_value) as u64)));
+            }
+            None => return Some(Err(anyhow!("'{}' is not a Bitcoin amount", string))),
+        };
+    }
+    None
+}
+
+fn check_m(string: &str) -> bool {
+    string.to_lowercase().contains("m")
+}
+
+fn check_k(string: &str) -> bool {
+    string.to_lowercase().contains("k")
+}
+
+fn void_denominations(string: &str) -> bool {
+    !(string.to_lowercase().ends_with("sat") | string.to_lowercase().ends_with("btc"))
```

Comment:
better to check that it *is* a `k` or and `M` before trying to parsing it as an `Amount` rather than checking it isn't one of the more common `Amount`s and then trying `k` or `M`.